packages/renderer/src/lib/task-manager/ProgressBar.svelte

```diff
@@ -29,12 +29,17 @@
 </style>
 
 <script lang="ts">
-export let progress: number | undefined;
-export let width: string = 'w-36';
-export let height: string = 'h-4';
+interface Props {
+  progress?: number;
+  width?: string;
+  height?: string;
+  class?: string;
```

Comment:
question: usually we let any other props being passed, we don't have a field for class

Reply:
then how do get the `class` prop in svelte 5 :thinking:  ? I took the solution from https://www.reddit.com/r/sveltejs/comments/1dodc08/svelte_5_class_prop/?rdt=40307

Reply:
maybe something like spread operator
```
let { progress, width = 'w-36', height = 'h-4', ...restProps }: Props = $props();
```

and then `{props.class}`

I suspect maybe it'll be missing the type so either
```
let { progress, width = 'w-36', height = 'h-4', ...restProps }: Props & {class? : string} = $props();
```

or allowing more attributes that we can set on divs:
```
let { progress, width = 'w-36', height = 'h-4', ...restProps }: Props & Record<string, unknown> = $props();
```

and using `{...restProps}` but in your class as you want to concatenate the value of class

`<div class="flex flex-row {restProps.class}">`

Reply:
Why would we want to allow untyped element to be added ? A component is suppose to have a determined number of props ? I am not really seeing the purpose to add `unknown` props ?

If we need to have more props that we want to propagate we should have them defined, and properly typed ?

Reply:
you can use & HTMLAttributes and {...restProps} meaning all attributes you can set on an html element can be propagated

Reply:
This is a bit complicated, as HTMLAttributes is a generic type. See definition bellow (from 'svelte/elements')

```ts
export interface HTMLAttributes<T extends EventTarget> extends AriaAttributes, DOMAttributes<T> {
```

We could try to work around by using the `HTMLElement` object, since `HTMLElement extend Node` and `Node extend EventTarget` and do something like

```ts
import type { HTMLAttributes } from 'svelte/elements';

interface Props extends HTMLAttributes<HTMLElement> {
     // my-custom component props
}

let { trackingId, ...restProps }: Props = $props();
```

restProps now can contains all HTML basic attributes such as `aria-*`, `role`, `title` etc.

This is the most typed and html friendly I can do, what do you think ?

Reply:
go for it :-)